pkgs/by-name/ol/olympus/unwrapped.nix

```diff
@@ -0,0 +1,106 @@
+{
+  lib,
+  fetchFromGitHub,
+  fetchzip,
+  buildDotnetModule,
+  dotnetCorePackages,
+  luajitPackages,
+  sqlite,
+  libarchive,
+  curl,
+  love,
+  xdg-utils,
+}:
+let
+  lua_cpath =
+    with luajitPackages;
+    lib.concatMapStringsSep ";" getLuaCPath [
+      (buildLuarocksPackage {
+        pname = "lsqlite3";
+        version = "0.9.6-1";
+        src = fetchzip {
+          url = "http://lua.sqlite.org/home/zip/lsqlite3_v096.zip";
+          hash = "sha256-Mq409A3X9/OS7IPI/KlULR6ZihqnYKk/mS/W/2yrGBg=";
+        };
+        buildInputs = [ sqlite.dev ];
+      })
+
+      lua-subprocess
+      nfd
+    ];
+
+  pname = "olympus";
+  phome = "$out/lib/${pname}";
+  # The following variables are to be updated by the update script.
+  version = "25.02.07.01";
+  buildId = "4624"; # IMPORTANT: This line is matched with regex in update.sh.
+  rev = "f4cd9dc973e68dc9b6c043941d5ab57f93b63ac4";
+in
+buildDotnetModule {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    inherit rev;
+    owner = "EverestAPI";
+    repo = "Olympus";
+    fetchSubmodules = true; # Required. See upstream's README.
+    hash = "sha256-I0tDqe7XvieL0kj8njzaNx3taY2VpFewi/SnYRCi4tk=";
+  };
+
+  nativeBuildInputs = [
+    libarchive # To create the .love file (zip format).
+  ];
+
+  nugetDeps = ./deps.json;
+  projectFile = "sharp/Olympus.Sharp.csproj";
+  executables = [ ];
+  installPath = "${placeholder "out"}/lib/${pname}/sharp";
+
+  # See the 'Dist: Update src/version.txt' step in azure-pipelines.yml from upstream.
+  preConfigure = ''
+    echo ${version}-nixos-${buildId}-${builtins.substring 0 5 rev} > src/version.txt
+  '';
+
+  # The script find-love is hacked to use love from nixpkgs.
+  # It is used to launch Loenn from Olympus.
+  # I assume --fused is so saves are properly made (https://love2d.org/wiki/love.filesystem).
+  preInstall = ''
+    mkdir -p ${phome}
+    makeWrapper ${lib.getExe love} ${phome}/find-love \
+      --add-flags "--fused"
+
+    mkdir -p $out/bin
+    makeWrapper ${phome}/find-love $out/bin/olympus \
+      --prefix LUA_CPATH ";" "${lua_cpath}" \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ curl ]}" \
+      --suffix PATH : "${lib.makeBinPath [ xdg-utils ]}" \
+      --add-flags ${phome}/olympus.love \
+      --set DOTNET_ROOT ${dotnetCorePackages.runtime_8_0}/share/dotnet
+
+    bsdtar --format zip --strip-components 1 -cf ${phome}/olympus.love src
+  '';
+
+  postInstall = ''
+    install -Dm644 lib-linux/olympus.desktop $out/share/applications/olympus.desktop
+    install -Dm644 src/data/icon.png $out/share/icons/hicolor/128x128/apps/olympus.png
+    install -Dm644 LICENSE $out/share/licenses/${pname}/LICENSE
+  '';
+
+  passthru = {
+    updateScript = ./update.sh;
+  };
```

Comment:
I think this should be moved to the wrapped package. Also I think we need to put the unwrapped package in the `passthru` of the wrapped package (otherwise the unwrapped package is inaccessible from a nix expression). Therefore, remove `passthru` in the unwrapped package, and write `passthru = { unwrapped = olympus-unwrapped; updateScript = ./update.sh; };` in the wrapped package.